app/models/spree/inventory_unit_decorator.rb

```diff
@@ -31,13 +31,13 @@ def increase_with_electronic_shipments(order, variant, quantity)
     # paraphrased from spree-core (create_units), with changes to assign
     # inventory units to electronic shipment or create new electronic shipment
     def create_electronic_units(order, variant, quantity)
-      shipment = order.shipments.electronic.first_or_create!
-
-      quantity.times do
-        order.inventory_units.create(
-          { variant: variant, state: "sold", shipment: shipment },
-          without_protection: true)
+      inventory_units = quantity.times.map do
+        order.inventory_units.create( { variant: variant, state: "sold" }, without_protection: true)
       end
+
+      shipment = order.shipments.electronic.first_or_initialize
+      shipment.inventory_units << inventory_units
```

Comment:
Thank you for comment.

[This test](https://github.com/degica/spree-license-key/pull/24/files#diff-00cfa2c46e33ca9c2ac0743ceaa83e89R53) will be fail if that line has previous same issue.